beacon-chain/sync/initial-sync/blocks_fetcher_test.go

```diff
@@ -0,0 +1,362 @@
+package initialsync
+
+import (
+	"context"
+	"sort"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/kevinms/leakybucket-go"
+	eth "github.com/prysmaticlabs/ethereumapis/eth/v1alpha1"
+	mock "github.com/prysmaticlabs/prysm/beacon-chain/blockchain/testing"
+	dbtest "github.com/prysmaticlabs/prysm/beacon-chain/db/testing"
+	p2pt "github.com/prysmaticlabs/prysm/beacon-chain/p2p/testing"
+	stateTrie "github.com/prysmaticlabs/prysm/beacon-chain/state"
+	p2ppb "github.com/prysmaticlabs/prysm/proto/beacon/p2p/v1"
+	"github.com/prysmaticlabs/prysm/shared/sliceutil"
+	"github.com/sirupsen/logrus"
+)
+
+func TestBlocksFetcher(t *testing.T) {
```

Comment:
One thing that could be very helpful for tests like this that are testing behavior of an entire service is to assert for logs. You can see an example here: https://github.com/prysmaticlabs/prysm/blob/b647ca5dd2f775742f1dc146ecfc58bed2aa2bd8/slasher/beaconclient/historical_data_retrieval_test.go#L78 we basically check easily if a log we want has occurred. You can use that to test edge cases or uncovered lines of your logic.